permissions/__main__.py

```diff
@@ -25,8 +25,11 @@ async def permission(request, username):
     else:
         post_data = await request.post() # request.json() crashes on empty data
         LOG.debug('POST DATA: %s', post_data)
-        requested_datasets = post_data.get('datasets', '').split(',')
-
+        try:
```

Comment:
`string.split(x)` is `[string]` if `x` is not found. So that try/catch is not necessary. It was already accounted for.

Reply:
It's not that you don't find `x`, is that now instead of getting "a,b,c,d" we get ["a", "b", "c", "d"]. Since now we use post data instead of the uri query.

Reply:
the .post() and .json() have already parsed the value as a list. JSON doesn't have sets.
So you should get separatly `d = post_data.get('datasets')` (and not use a default value), and use:
```
if d is None:
      return []
if isinstance(d, list):
     return d
else:
    d.split(',') # maybe add an assert( isinstance(d, str))
```

Reply:
Change `return []` and `return d` to `return set()` and `return set(d)` in the case of sets.